// @workos-inc/node/lib/index.worker.cjs
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const require_factory = require("./factory-DwW2quUn.cjs");
//#region src/index.worker.ts
var WorkOSWorker = class extends require_factory.WorkOS {
	/** @override */
	createHttpClient(options, userAgent) {
		const headers = { "User-Agent": userAgent };
		const configHeaders = options.config?.headers;
		if (configHeaders && typeof configHeaders === "object" && !Array.isArray(configHeaders) && !(configHeaders instanceof Headers)) Object.assign(headers, configHeaders);
		if (this.key) headers["Authorization"] = `Bearer ${this.key}`;
		return new require_factory.FetchHttpClient(this.baseURL, {
			...options.config,
			headers
		});
	}
	/** @override */
	createWebhookClient() {
		return new require_factory.Webhooks(this);
	}
	getCryptoProvider() {
		return new require_factory.SubtleCryptoProvider();
	}
	/** @override */
	createActionsClient() {
		return new require_factory.Actions(new require_factory.SubtleCryptoProvider());
	}
	/** @override */
	emitWarning(warning) {
		return console.warn(`WorkOS: ${warning}`);
	}
};
//#endregion
exports.ApiKeyRequiredException = require_factory.ApiKeyRequiredException;
exports.AuthenticateWithSessionCookieFailureReason = require_factory.AuthenticateWithSessionCookieFailureReason;
exports.AuthenticationException = require_factory.AuthenticationException;
exports.AutoPaginatable = require_factory.AutoPaginatable;
exports.BadRequestException = require_factory.BadRequestException;
exports.ConflictException = require_factory.ConflictException;
exports.ConnectionType = require_factory.ConnectionType;
exports.DomainDataState = require_factory.DomainDataState;
exports.GenerateLinkIntent = require_factory.GenerateLinkIntent;
exports.GenericServerException = require_factory.GenericServerException;
exports.NoApiKeyProvidedException = require_factory.NoApiKeyProvidedException;
exports.NotFoundException = require_factory.NotFoundException;
exports.OauthException = require_factory.OauthException;
exports.OrganizationDomainState = require_factory.OrganizationDomainState;
exports.OrganizationDomainVerificationStrategy = require_factory.OrganizationDomainVerificationStrategy;
exports.PKCE = require_factory.PKCE;
exports.RateLimitExceededException = require_factory.RateLimitExceededException;
exports.RefreshSessionFailureReason = require_factory.RefreshSessionFailureReason;
exports.SignatureVerificationException = require_factory.SignatureVerificationException;
exports.UnauthorizedException = require_factory.UnauthorizedException;
exports.UnprocessableEntityException = require_factory.UnprocessableEntityException;
exports.WorkOS = WorkOSWorker;
exports.createWorkOS = require_factory.createWorkOS;
exports.isAuthenticationErrorData = require_factory.isAuthenticationErrorData;
exports.serializeRevokeSessionOptions = require_factory.serializeRevokeSessionOptions;
