// @workos-inc/node/lib/index.worker.mjs
import { A as FetchHttpClient, C as NoApiKeyProvidedException, D as isAuthenticationErrorData, E as AuthenticationException, O as GenericServerException, S as NotFoundException, T as BadRequestException, _ as UnprocessableEntityException, a as DomainDataState, b as RateLimitExceededException, d as serializeRevokeSessionOptions, f as AuthenticateWithSessionCookieFailureReason, g as Actions, h as AutoPaginatable, i as OrganizationDomainVerificationStrategy, j as SubtleCryptoProvider, k as ApiKeyRequiredException, m as Webhooks, n as ConnectionType, o as GenerateLinkIntent, p as PKCE, r as OrganizationDomainState, s as WorkOS, t as createWorkOS, u as RefreshSessionFailureReason, v as UnauthorizedException, w as ConflictException, x as OauthException, y as SignatureVerificationException } from "./factory-BjWO1KkE.mjs";
//#region src/index.worker.ts
var WorkOSWorker = class extends WorkOS {
	/** @override */
	createHttpClient(options, userAgent) {
		const headers = { "User-Agent": userAgent };
		const configHeaders = options.config?.headers;
		if (configHeaders && typeof configHeaders === "object" && !Array.isArray(configHeaders) && !(configHeaders instanceof Headers)) Object.assign(headers, configHeaders);
		if (this.key) headers["Authorization"] = `Bearer ${this.key}`;
		return new FetchHttpClient(this.baseURL, {
			...options.config,
			headers
		});
	}
	/** @override */
	createWebhookClient() {
		return new Webhooks(this);
	}
	getCryptoProvider() {
		return new SubtleCryptoProvider();
	}
	/** @override */
	createActionsClient() {
		return new Actions(new SubtleCryptoProvider());
	}
	/** @override */
	emitWarning(warning) {
		return console.warn(`WorkOS: ${warning}`);
	}
};
//#endregion
export { ApiKeyRequiredException, AuthenticateWithSessionCookieFailureReason, AuthenticationException, AutoPaginatable, BadRequestException, ConflictException, ConnectionType, DomainDataState, GenerateLinkIntent, GenericServerException, NoApiKeyProvidedException, NotFoundException, OauthException, OrganizationDomainState, OrganizationDomainVerificationStrategy, PKCE, RateLimitExceededException, RefreshSessionFailureReason, SignatureVerificationException, UnauthorizedException, UnprocessableEntityException, WorkOSWorker as WorkOS, createWorkOS, isAuthenticationErrorData, serializeRevokeSessionOptions };
