// @workos-inc/node/lib/index.cjs
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const require_factory = require("./factory-DwW2quUn.cjs");
//#region src/groups/interfaces/user-organization-membership-base-list-data-status.interface.ts
const UserOrganizationMembershipBaseListDataStatus = {
	Active: "active",
	Inactive: "inactive",
	Pending: "pending"
};
//#endregion
//#region src/radar/interfaces/radar-list-action.interface.ts
const RadarListAction = {
	Block: "block",
	Allow: "allow"
};
//#endregion
//#region src/radar/interfaces/radar-list-type.interface.ts
const RadarListType = {
	IpAddress: "ip_address",
	Domain: "domain",
	Email: "email",
	Device: "device",
	UserAgent: "user_agent",
	DeviceFingerprint: "device_fingerprint",
	Country: "country"
};
//#endregion
//#region src/radar/interfaces/radar-standalone-assess-request-action.interface.ts
const RadarStandaloneAssessRequestAction = {
	SignUp: "sign-up",
	SignIn: "sign-in"
};
//#endregion
//#region src/radar/interfaces/radar-standalone-assess-request-auth-method.interface.ts
const RadarStandaloneAssessRequestAuthMethod = {
	Password: "Password",
	Passkey: "Passkey",
	Authenticator: "Authenticator",
	SmsOtp: "SMS_OTP",
	EmailOtp: "Email_OTP",
	Social: "Social",
	SSO: "SSO",
	Other: "Other"
};
//#endregion
//#region src/radar/interfaces/radar-standalone-response-blocklist-type.interface.ts
const RadarStandaloneResponseBlocklistType = {
	IpAddress: "ip_address",
	Domain: "domain",
	Email: "email",
	Device: "device",
	UserAgent: "user_agent",
	DeviceFingerprint: "device_fingerprint",
	Country: "country"
};
//#endregion
//#region src/radar/interfaces/radar-standalone-response-control.interface.ts
const RadarStandaloneResponseControl = {
	BotDetection: "bot_detection",
	BruteForceAttack: "brute_force_attack",
	DomainSignUpRateLimit: "domain_sign_up_rate_limit",
	ImpossibleTravel: "impossible_travel",
	RepeatSignUp: "repeat_sign_up",
	StaleAccount: "stale_account",
	UnrecognizedDevice: "unrecognized_device",
	Restriction: "restriction"
};
//#endregion
//#region src/radar/interfaces/radar-standalone-response-verdict.interface.ts
const RadarStandaloneResponseVerdict = {
	Allow: "allow",
	Block: "block",
	Challenge: "challenge"
};
//#endregion
//#region src/vault/interfaces/vault-order.interface.ts
const VaultOrder = {
	Asc: "asc",
	Desc: "desc"
};
//#endregion
//#region src/webhooks/interfaces/create-webhook-endpoint-events.interface.ts
const CreateWebhookEndpointEvents = {
	AuthenticationEmailVerificationSucceeded: "authentication.email_verification_succeeded",
	AuthenticationMagicAuthFailed: "authentication.magic_auth_failed",
	AuthenticationMagicAuthSucceeded: "authentication.magic_auth_succeeded",
	AuthenticationMfaSucceeded: "authentication.mfa_succeeded",
	AuthenticationOAuthFailed: "authentication.oauth_failed",
	AuthenticationOAuthSucceeded: "authentication.oauth_succeeded",
	AuthenticationPasswordFailed: "authentication.password_failed",
	AuthenticationPasswordSucceeded: "authentication.password_succeeded",
	AuthenticationPasskeyFailed: "authentication.passkey_failed",
	AuthenticationPasskeySucceeded: "authentication.passkey_succeeded",
	AuthenticationSSOFailed: "authentication.sso_failed",
	AuthenticationSSOStarted: "authentication.sso_started",
	AuthenticationSSOSucceeded: "authentication.sso_succeeded",
	AuthenticationSSOTimedOut: "authentication.sso_timed_out",
	AuthenticationRadarRiskDetected: "authentication.radar_risk_detected",
	ApiKeyCreated: "api_key.created",
	ApiKeyRevoked: "api_key.revoked",
	ConnectionActivated: "connection.activated",
	ConnectionDeactivated: "connection.deactivated",
	ConnectionSAMLCertificateRenewalRequired: "connection.saml_certificate_renewal_required",
	ConnectionSAMLCertificateRenewed: "connection.saml_certificate_renewed",
	ConnectionDeleted: "connection.deleted",
	DsyncActivated: "dsync.activated",
	DsyncDeleted: "dsync.deleted",
	DsyncGroupCreated: "dsync.group.created",
	DsyncGroupDeleted: "dsync.group.deleted",
	DsyncGroupUpdated: "dsync.group.updated",
	DsyncGroupUserAdded: "dsync.group.user_added",
	DsyncGroupUserRemoved: "dsync.group.user_removed",
	DsyncUserCreated: "dsync.user.created",
	DsyncUserDeleted: "dsync.user.deleted",
	DsyncUserUpdated: "dsync.user.updated",
	EmailVerificationCreated: "email_verification.created",
	GroupCreated: "group.created",
	GroupDeleted: "group.deleted",
	GroupMemberAdded: "group.member_added",
	GroupMemberRemoved: "group.member_removed",
	GroupUpdated: "group.updated",
	FlagCreated: "flag.created",
	FlagDeleted: "flag.deleted",
	FlagUpdated: "flag.updated",
	FlagRuleUpdated: "flag.rule_updated",
	InvitationAccepted: "invitation.accepted",
	InvitationCreated: "invitation.created",
	InvitationResent: "invitation.resent",
	InvitationRevoked: "invitation.revoked",
	MagicAuthCreated: "magic_auth.created",
	OrganizationCreated: "organization.created",
	OrganizationDeleted: "organization.deleted",
	OrganizationUpdated: "organization.updated",
	OrganizationDomainCreated: "organization_domain.created",
	OrganizationDomainDeleted: "organization_domain.deleted",
	OrganizationDomainUpdated: "organization_domain.updated",
	OrganizationDomainVerified: "organization_domain.verified",
	OrganizationDomainVerificationFailed: "organization_domain.verification_failed",
	PasswordResetCreated: "password_reset.created",
	PasswordResetSucceeded: "password_reset.succeeded",
	UserCreated: "user.created",
	UserUpdated: "user.updated",
	UserDeleted: "user.deleted",
	OrganizationMembershipCreated: "organization_membership.created",
	OrganizationMembershipDeleted: "organization_membership.deleted",
	OrganizationMembershipUpdated: "organization_membership.updated",
	RoleCreated: "role.created",
	RoleDeleted: "role.deleted",
	RoleUpdated: "role.updated",
	OrganizationRoleCreated: "organization_role.created",
	OrganizationRoleDeleted: "organization_role.deleted",
	OrganizationRoleUpdated: "organization_role.updated",
	PermissionCreated: "permission.created",
	PermissionDeleted: "permission.deleted",
	PermissionUpdated: "permission.updated",
	PipesConnectedAccountConnected: "pipes.connected_account.connected",
	PipesConnectedAccountDisconnected: "pipes.connected_account.disconnected",
	PipesConnectedAccountReauthorizationNeeded: "pipes.connected_account.reauthorization_needed",
	SessionCreated: "session.created",
	SessionRevoked: "session.revoked",
	WaitlistUserApproved: "waitlist_user.approved",
	WaitlistUserCreated: "waitlist_user.created",
	WaitlistUserDenied: "waitlist_user.denied"
};
//#endregion
//#region src/webhooks/interfaces/update-webhook-endpoint-events.interface.ts
const UpdateWebhookEndpointEvents = {
	AuthenticationEmailVerificationSucceeded: "authentication.email_verification_succeeded",
	AuthenticationMagicAuthFailed: "authentication.magic_auth_failed",
	AuthenticationMagicAuthSucceeded: "authentication.magic_auth_succeeded",
	AuthenticationMfaSucceeded: "authentication.mfa_succeeded",
	AuthenticationOAuthFailed: "authentication.oauth_failed",
	AuthenticationOAuthSucceeded: "authentication.oauth_succeeded",
	AuthenticationPasswordFailed: "authentication.password_failed",
	AuthenticationPasswordSucceeded: "authentication.password_succeeded",
	AuthenticationPasskeyFailed: "authentication.passkey_failed",
	AuthenticationPasskeySucceeded: "authentication.passkey_succeeded",
	AuthenticationSSOFailed: "authentication.sso_failed",
	AuthenticationSSOStarted: "authentication.sso_started",
	AuthenticationSSOSucceeded: "authentication.sso_succeeded",
	AuthenticationSSOTimedOut: "authentication.sso_timed_out",
	AuthenticationRadarRiskDetected: "authentication.radar_risk_detected",
	ApiKeyCreated: "api_key.created",
	ApiKeyRevoked: "api_key.revoked",
	ConnectionActivated: "connection.activated",
	ConnectionDeactivated: "connection.deactivated",
	ConnectionSAMLCertificateRenewalRequired: "connection.saml_certificate_renewal_required",
	ConnectionSAMLCertificateRenewed: "connection.saml_certificate_renewed",
	ConnectionDeleted: "connection.deleted",
	DsyncActivated: "dsync.activated",
	DsyncDeleted: "dsync.deleted",
	DsyncGroupCreated: "dsync.group.created",
	DsyncGroupDeleted: "dsync.group.deleted",
	DsyncGroupUpdated: "dsync.group.updated",
	DsyncGroupUserAdded: "dsync.group.user_added",
	DsyncGroupUserRemoved: "dsync.group.user_removed",
	DsyncUserCreated: "dsync.user.created",
	DsyncUserDeleted: "dsync.user.deleted",
	DsyncUserUpdated: "dsync.user.updated",
	EmailVerificationCreated: "email_verification.created",
	GroupCreated: "group.created",
	GroupDeleted: "group.deleted",
	GroupMemberAdded: "group.member_added",
	GroupMemberRemoved: "group.member_removed",
	GroupUpdated: "group.updated",
	FlagCreated: "flag.created",
	FlagDeleted: "flag.deleted",
	FlagUpdated: "flag.updated",
	FlagRuleUpdated: "flag.rule_updated",
	InvitationAccepted: "invitation.accepted",
	InvitationCreated: "invitation.created",
	InvitationResent: "invitation.resent",
	InvitationRevoked: "invitation.revoked",
	MagicAuthCreated: "magic_auth.created",
	OrganizationCreated: "organization.created",
	OrganizationDeleted: "organization.deleted",
	OrganizationUpdated: "organization.updated",
	OrganizationDomainCreated: "organization_domain.created",
	OrganizationDomainDeleted: "organization_domain.deleted",
	OrganizationDomainUpdated: "organization_domain.updated",
	OrganizationDomainVerified: "organization_domain.verified",
	OrganizationDomainVerificationFailed: "organization_domain.verification_failed",
	PasswordResetCreated: "password_reset.created",
	PasswordResetSucceeded: "password_reset.succeeded",
	UserCreated: "user.created",
	UserUpdated: "user.updated",
	UserDeleted: "user.deleted",
	OrganizationMembershipCreated: "organization_membership.created",
	OrganizationMembershipDeleted: "organization_membership.deleted",
	OrganizationMembershipUpdated: "organization_membership.updated",
	RoleCreated: "role.created",
	RoleDeleted: "role.deleted",
	RoleUpdated: "role.updated",
	OrganizationRoleCreated: "organization_role.created",
	OrganizationRoleDeleted: "organization_role.deleted",
	OrganizationRoleUpdated: "organization_role.updated",
	PermissionCreated: "permission.created",
	PermissionDeleted: "permission.deleted",
	PermissionUpdated: "permission.updated",
	PipesConnectedAccountConnected: "pipes.connected_account.connected",
	PipesConnectedAccountDisconnected: "pipes.connected_account.disconnected",
	PipesConnectedAccountReauthorizationNeeded: "pipes.connected_account.reauthorization_needed",
	SessionCreated: "session.created",
	SessionRevoked: "session.revoked",
	WaitlistUserApproved: "waitlist_user.approved",
	WaitlistUserCreated: "waitlist_user.created",
	WaitlistUserDenied: "waitlist_user.denied"
};
//#endregion
//#region src/webhooks/interfaces/update-webhook-endpoint-status.interface.ts
const UpdateWebhookEndpointStatus = {
	Enabled: "enabled",
	Disabled: "disabled"
};
//#endregion
//#region src/webhooks/interfaces/webhook-endpoint-status.interface.ts
const WebhookEndpointStatus = {
	Enabled: "enabled",
	Disabled: "disabled"
};
//#endregion
//#region src/widgets/interfaces/widget-session-token-scopes.interface.ts
const WidgetSessionTokenScopes = {
	WidgetsUsersTableManage: "widgets:users-table:manage",
	WidgetsDomainVerificationManage: "widgets:domain-verification:manage",
	WidgetsSSOManage: "widgets:sso:manage",
	WidgetsApiKeysManage: "widgets:api-keys:manage",
	WidgetsDsyncManage: "widgets:dsync:manage",
	WidgetsAuditLogStreamingManage: "widgets:audit-log-streaming:manage"
};
//#endregion
//#region src/index.ts
var WorkOSNode = class extends require_factory.WorkOS {
	/** @override */
	createHttpClient(options, userAgent) {
		const headers = {};
		const configHeaders = options.config?.headers;
		if (configHeaders) if (configHeaders instanceof Headers) configHeaders.forEach((v, k) => headers[k] = v);
		else if (Array.isArray(configHeaders)) configHeaders.forEach(([k, v]) => headers[k] = v);
		else Object.assign(headers, configHeaders);
		headers["User-Agent"] = userAgent;
		if (this.key) headers["Authorization"] = `Bearer ${this.key}`;
		const opts = {
			...options.config,
			timeout: options.timeout,
			headers
		};
		return new require_factory.FetchHttpClient(this.baseURL, opts, options.fetchFn);
	}
	/** @override */
	createWebhookClient() {
		return new require_factory.Webhooks(this);
	}
	getCryptoProvider() {
		return new require_factory.SubtleCryptoProvider();
	}
	/** @override */
	createActionsClient() {
		return new require_factory.Actions(this.getCryptoProvider());
	}
	/** @override */
	emitWarning(warning) {
		return process.emitWarning(warning, "WorkOS");
	}
};
//#endregion
exports.ApiKeyRequiredException = require_factory.ApiKeyRequiredException;
exports.AuthenticateWithSessionCookieFailureReason = require_factory.AuthenticateWithSessionCookieFailureReason;
exports.AuthenticationException = require_factory.AuthenticationException;
exports.AutoPaginatable = require_factory.AutoPaginatable;
exports.BadRequestException = require_factory.BadRequestException;
exports.ConflictException = require_factory.ConflictException;
exports.ConnectionType = require_factory.ConnectionType;
exports.CookieSession = require_factory.CookieSession;
exports.CreateWebhookEndpointEvents = CreateWebhookEndpointEvents;
exports.DomainDataState = require_factory.DomainDataState;
exports.FeatureFlagsRuntimeClient = require_factory.FeatureFlagsRuntimeClient;
exports.GenerateLinkIntent = require_factory.GenerateLinkIntent;
exports.GenericServerException = require_factory.GenericServerException;
exports.NoApiKeyProvidedException = require_factory.NoApiKeyProvidedException;
exports.NotFoundException = require_factory.NotFoundException;
exports.OauthException = require_factory.OauthException;
exports.OrganizationDomainState = require_factory.OrganizationDomainState;
exports.OrganizationDomainVerificationStrategy = require_factory.OrganizationDomainVerificationStrategy;
exports.PKCE = require_factory.PKCE;
exports.RadarListAction = RadarListAction;
exports.RadarListType = RadarListType;
exports.RadarStandaloneAssessRequestAction = RadarStandaloneAssessRequestAction;
exports.RadarStandaloneAssessRequestAuthMethod = RadarStandaloneAssessRequestAuthMethod;
exports.RadarStandaloneResponseBlocklistType = RadarStandaloneResponseBlocklistType;
exports.RadarStandaloneResponseControl = RadarStandaloneResponseControl;
exports.RadarStandaloneResponseVerdict = RadarStandaloneResponseVerdict;
exports.RateLimitExceededException = require_factory.RateLimitExceededException;
exports.RefreshSessionFailureReason = require_factory.RefreshSessionFailureReason;
exports.SignatureVerificationException = require_factory.SignatureVerificationException;
exports.UnauthorizedException = require_factory.UnauthorizedException;
exports.UnprocessableEntityException = require_factory.UnprocessableEntityException;
exports.UpdateWebhookEndpointEvents = UpdateWebhookEndpointEvents;
exports.UpdateWebhookEndpointStatus = UpdateWebhookEndpointStatus;
exports.UserOrganizationMembershipBaseListDataStatus = UserOrganizationMembershipBaseListDataStatus;
exports.VaultOrder = VaultOrder;
exports.WebhookEndpointStatus = WebhookEndpointStatus;
exports.WidgetSessionTokenScopes = WidgetSessionTokenScopes;
exports.WorkOS = WorkOSNode;
exports.createWorkOS = require_factory.createWorkOS;
exports.isAuthenticationErrorData = require_factory.isAuthenticationErrorData;
exports.serializeRevokeSessionOptions = require_factory.serializeRevokeSessionOptions;
